pages/api/retrieval.ts

```diff
@@ -0,0 +1,112 @@
+  if (!maxRecords || parseInt(maxRecords as string) < 0 || parseInt(maxRecords as string) >= 11) {
+    return res.status(400).json({ error: "Invalid payload value found maxRecords" });
+  } else {
+    //Get formID from the bearer token
+    const formID = bearerTokenPayload?.formID;
+    if (!formID) return res.status(404).json({ error: "Bad Request" });
+    //A list to store submissionIDs
+    let submissionIDlist: (string | undefined)[] | undefined = [];
+    try {
+      //Create dynamodb client
+      const db = new DynamoDBClient({ region: process.env.AWS_REGION ?? "ca-central-1" });
+      //Create form responses db param
+      //NB: A filter expression is applied after a Query finishes, but before the results are returned.
+      //Therefore, a Query consumes the same amount of read capacity, regardless of whether a filter expression is present.
+      const getItemsDbParams = {
+        TableName: "Vault",
+        IndexName: "retrieved-index",
+        Limit: parseInt(maxRecords as string),
+        //Cannot use partitin key or sort key such as formID and Retrieved attribute in keyConditionExpression simultaneously
+        KeyConditionExpression: "Retrieved = :zero",
+        ExpressionAttributeValues: { ":formID": { S: formID }, ":zero": { N: "0" } },
+        //A filter expression cannot contain partition key or sort key attributes.
+        //You need to specify those attributes in the key condition expression, not the filter expression.
+        FilterExpression: "FormID = :formID",
+        ProjectionExpression: "FormID,SubmissionID,FormSubmission,Retrieved",
+      };
+      //Get form's responses for formID
+      const formResponses = await db.send(new QueryCommand(getItemsDbParams));
+      //Collecting items submissionIDs for logging and updating items.
+      submissionIDlist = formResponses?.Items?.map((response) => {
+        return response.SubmissionID.S;
+      });
+      if (formResponses && formResponses.Count && formResponses.Count > 0) {
+        //Preparing a query to set records's attribute Retrieved to 1.
+        for (const submissionID of submissionIDlist as string[]) {
+          //Create a new statement object
+          const updateItem = {
+            TableName: "Vault",
+            Key: {
+              SubmissionID: {
+                S: submissionID,
+              },
+              FormID: {
+                S: formID,
+              },
+            },
+            UpdateExpression: "SET Retrieved = :retrieved",
+            ConditionExpression: "SubmissionID = :submission",
+            ExpressionAttributeValues: {
+              ":retrieved": { N: "1" },
+              ":submission": { S: submissionID },
+            },
+            ReturnValues: "NONE",
+          };
+          //Update one item at the time
+          await db.send(new UpdateItemCommand(updateItem));
+        }
+      }
+      //Return form data
+      return res.status(200).json({ responses: formResponses.Items });
```

Comment:
It would probably be better to flatten our JSON structure as opposed to having 

```JSON
 {
  "someKey": {
      "S": "value"
}
}
```

You can probably do this via a map function

Reply:
To keep this PR short , I'll address this improvement in a separate card.